examples/cloud-tomee-azure/README.adoc

```diff
@@ -0,0 +1,101 @@
+
+== Azure Setup
+
+In order for the Azure plugin to work, you will need to have an Azure account and add a subscription to it.
+Then, on yor development machine, install the Azure command-line interface (CLI) and authenticate with the command
+line, before you can deploy your application.
+
+- Create an Azure Account, if you don't have one, at https://azure.microsoft.com/en-us
+- Use the free option, if available or https://portal.azure.com/#blade/Microsoft_Azure_Billing/SubscriptionsBlade[add a subscription].
+- https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest[Install] the Azure (CLI) according
+to operating system of the computer you are using to develop.
+
+- Finally you can setup your development computer.
+
+=== Login into Azure
+
+`az login`
+
+The result:
+
+----
+[
+   {
+     "cloudName": "AzureCloud",
+     "id": "aaaaaaaa-aaaa-aaaa-aaaaa-aaaaaaaaaa",
+     "isDefault": true,
+     "name": "Pay-As-You-Go",
+     "state": "Enabled",
+     "tenantId": "bbbbbbb-bbbbb-bbbb-bbbbb-bbbbbbbbbbb",
+     "user": {
+       "name": "<your azure account's email>",
+       "type": "user"
+     }
+   }
+ ]
+----
+The TenantId is someone that can register and manage apps, yourself. You will need that for later.
+
+
+=== Create a service principal
+An Azure service principal is a security identity used by user-created apps, services,
+and automation tools to access specific Azure resources:
+
+`az ad sp create-for-rbac --name ServicePrincipalName --password <Choose a strong password>`
+
+----
+az ad sp create-for-rbac --name  http://cloud-tomee-azure --password <password for this app>
+
+{
+  "appId": "cccccccc-cccc-cccc-cccc-ccccccccccccccc",
+  "displayName": "cloud-tomee-azure",
+  "name": "http://cloud-tomee-azure",
+  "password": "<password for this app>",
+  "tenant": "bbbbbbb-bbbbb-bbbb-bbbbb-bbbbbbbbbbb"
+}
+----
+The ServicePrincipalName will also be set as subdomain name. In this case "http://cloud-tomee-azure".
+The appId is the identification of the app service.
+
+=== Configure Maven
+
+You could continue just using the Azure CLI for authentication but we can also do it permanently with maven.
+
+In that case we need to edit Maven's settings.xml so the azure-webapp-maven-plugin can authenticate against Azure:
+
+You can add a new server to `~/.m2/settings.xml` like this:
+----
+<server>
+  <id>azure-auth</id>
+  <configuration>
+     <client>cccccccc-cccc-cccc-cccc-ccccccccccccccc</client>
+     <tenant>bbbbbbb-bbbbb-bbbb-bbbbb-bbbbbbbbbbb</tenant>
+     <key><password for this app></key>
+     <environment>AZURE</environment>
+   </configuration>
+</server>
+----
+That's it. You can now build the example and deploy it to Azure just using Maven:
+
+`mvn clean install -Pazure-single-jar azure-webapp:deploy`
+
+The azure-webapp is explicitly invoked because it relies on you Azure account. The standard TomEE build will not use an Azure account.
```

Comment:
Here I would like to see the URL deployed on Azur cloud. I know the URL may change from user to user but at least the basic format.

Reply:
I initially thought that maven configuration in this example was wrong, and I tried to  replace: 
`<client>cccccccc-cccc-cccc-cccc-ccccccccccccccc</client>` with `<cappID>cccccccc-cccc-cccc-cccc-ccccccccccccccc</appID>`. 
The above change produced and error:
[ERROR] Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:1.5.2:deploy (default-cli) on project cloud-tomee-azure: Failed to authenticate with Azure. Please check your configuration. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

So I revert my change and let the maven configuration as described in this article, then I execute:
`mvn clean install -Pazure-single-jar azure-webapp:deploy` but it gives me now this error:

```
[INFO]
[INFO] --- azure-webapp-maven-plugin:1.5.2:deploy (default-cli) @ cloud-tomee-azure ---
[INFO] Authenticate with ServerId: azure-auth
[INFO] [Correlation ID: 1bec1ede-0f81-4905-babd-424a8d4d8f91] Instance discovery was successful
[INFO] Target Web App doesn't exist. Creating a new one...
[INFO] Creating App Service Plan 'ServicePlan46315a21-88f9-4150'...
[INFO] Successfully created App Service Plan.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:53 min
[INFO] Finished at: 2019-02-05T10:23:04-06:00
[INFO] Final Memory: 85M/1630M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:1.5.2:deploy (default-cli) on project cloud-tomee-azure: Website with given name cloud-tomee-azure already exists. OnError while emitting onNext value: retrofit2.Response.class -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

I tried to see if there is a `redeploy` or `undeploy` but it seems not available:
https://github.com/Microsoft/azure-maven-plugins/tree/master/azure-webapp-maven-plugin#goals

I went to the Azure web UI and I found my `resource-group` called `tomee-group` but I don't see inside the group any webapp deployed :/

I had to delete the  resource group tomee-group, an run again `mvn clean install -Pazure-single-jar azure-webapp:deploy` but I got the same issue:
```
[INFO] --- azure-webapp-maven-plugin:1.5.2:deploy (default-cli) @ cloud-tomee-azure ---
[INFO] Authenticate with ServerId: azure-auth
[INFO] [Correlation ID: bc22f4aa-067b-419e-9be0-2387f4579b4e] Instance discovery was successful
[INFO] Target Web App doesn't exist. Creating a new one...
[INFO] Creating App Service Plan 'ServicePlana6463b1d-1ce8-408b'...
[INFO] Successfully created App Service Plan.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 42.301 s
[INFO] Finished at: 2019-02-05T10:49:50-06:00
[INFO] Final Memory: 73M/924M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:1.5.2:deploy (default-cli) on project cloud-tomee-azure: Website with given name cloud-tomee-azure already exists. OnError while emitting onNext value: retrofit2.Response.class -> [Help 1]
```

Reply:
@cesarhernandezgt  Have you sign up for the pay as you go plan?

Reply:
I think those sub-domains might need to be unique... can you try with a Different app name? not cloud-tomee-azure.

Reply:
Yes, the issue was the unique sub-domain, see my comment: https://github.com/apache/tomee/pull/378/files#r254044507

Reply:
> Here I would like to see the URL deployed on Azur cloud. I know the URL may change from user to user but at least the basic format.

This is how a url looks like once the deployemt is done, I think this should be included before the == Notes section.
```
https://<TheappNameYouUpdatedOnPomFile>.azurewebsites.net/cloud-tomee-azure-8.0.0-SNAPSHOT/echo/hi!
```

Reply:
done